### PR TITLE
Remove Makefile and move test scripts in package.json.

### PR DESCRIPTION
**Pull request by XhmikosR**
@pierreinglebert: please have a look and let me know if you want me to make any changes.

Notes:

1. I still think we can get rid of istanbul etc devDependencies if you only plan to run those via Travis
2. Not sure if all of those scripts are needed; I just moved them
3. We should make use of Travis cache

### CODE REVIEW

**Comment by pierreinglebert** [comment]
nope, I run istanbul on my dev :) And I prefer to have coveralls version in the devDependencies to be able to track it with david-dm

travis cache is another "issue".

**Comment by XhmikosR** [comment]
I added that in this branch; only node_modules for now.

**Comment by XhmikosR** [comment]
Not sure how to proceed; we both made similar PRs but not identical :/

**Comment by XhmikosR** [comment]
@pierreinglebert: please check this branch now and let me know if you need me to make any other changes :)

File: package.json [hunk 1]
```diff
@@ -52,8 +52,14 @@
...
   },
   "scripts": {
+    "build-source": "node-pre-gyp rebuild",
+    "coveralls": "coveralls < ./coverage/lcov.info",
     "install": "node-pre-gyp install --fallback-to-build --build-from-source",
-    "test": "make test"
+    "lint": "jshint .",
+    "mocha": "mocha test",
+    "test-cov": "istanbul cover ./node_modules/mocha/bin/_mocha -- -R spec test",
+    "test-coveralls": "npm run test-cov && istanbul-coveralls",
```
Comment (pierreinglebert) [inline package.json]: No more necessary

**Comment by pierreinglebert** [comment]
After test-coveralls, are we good ?

**Comment by XhmikosR** [comment]
Yes. Do you mind me squashing the patches?

**Comment by pierreinglebert** [comment]
nope, it's clearer that way :)

**Comment by XhmikosR** [comment]
Yeah, I was stuck for a moment; now it should be OK.

**Pull request merged**